command/aarelayer/service/aa_relayer_service.go

```diff
@@ -0,0 +1,161 @@
+package service
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/0xPolygon/polygon-edge/txrelayer"
+	"github.com/0xPolygon/polygon-edge/types"
+	"github.com/umbracle/ethgo"
+)
+
+// AARelayerService pulls transaction from pool one at the time and sends it to relayer
+type AARelayerService struct {
+	pool      AAPool
+	state     AATxState
+	txRelayer txrelayer.TxRelayer
+	key       ethgo.Key
+	pullTime  time.Duration
+}
+
+func NewAARelayerService(
+	txRelayer txrelayer.TxRelayer,
+	pool AAPool,
+	state AATxState,
+	key ethgo.Key,
+	opts ...TxRelayerOption) *AARelayerService {
+	service := &AARelayerService{
+		txRelayer: txRelayer,
+		pool:      pool,
+		state:     state,
+		key:       key,
+		pullTime:  time.Millisecond * 5000,
+	}
+
+	for _, opt := range opts {
+		opt(service)
+	}
+
+	return service
+}
+
+func (rs *AARelayerService) Start(ctx context.Context) {
+	ticker := time.NewTicker(rs.pullTime)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-ticker.C:
+			stateTx := rs.pool.Pop()
+			if stateTx != nil { // there is something to process
+				go func() {
+					if err := rs.executeJob(ctx, stateTx); err != nil {
+						// TODO: log error in file not just fmt.Println
+						fmt.Println(err)
+					}
+				}()
+			}
+		}
+	}
+}
+
+func (rs *AARelayerService) executeJob(ctx context.Context, stateTx *AAStateTransaction) error {
+	var (
+		netErr net.Error
+		tx     = rs.makeEthgoTransaction(stateTx)
+	)
+
+	hash, err := rs.txRelayer.SendTransactionWithoutReceipt(tx, rs.key)
+	// if its network error return tx back to the pool
+	if errors.As(err, &netErr) {
+		rs.pool.Push(stateTx)
+
+		return err
+	} else if err != nil {
+		errstr := err.Error()
+		stateTx.Error = &errstr
+
+		if errUpdate := rs.state.Update(stateTx); errUpdate != nil {
+			errstr = errUpdate.Error()
+
+			return fmt.Errorf("error while getting nonce for state tx = %s, err = %w, update error = %s",
+				stateTx.ID, err, errstr)
+		}
+
+		return fmt.Errorf("error while getting nonce for state tx = %s, err = %w", stateTx.ID, err)
+	}
+
+	stateTx.Status = StatusQueued
+	if err := rs.state.Update(stateTx); err != nil {
+		// TODO: log error but do not return
+		fmt.Printf("error while updating state tx = %s after sending it, err = %v", stateTx.ID, err)
+	}
+
+	select {
+	case <-ctx.Done():
+		return nil
+	default:
+	}
+
+	recipt, err := rs.txRelayer.WaitForReceipt(ctx, hash)
```

Comment:
At this point, the relayer is only an abstraction to send a transaction (`SendTransactionWithoutReceipt`) and wait for the receipt (`WaitForReceipt`). You are not leveraging the utilities of the relayer (i.e. fire and forget the execution of a transaction). Thus, you can use directly the `jsonrpc.Client` package to send and wait for the receipt without complicating the `txrelayer`.